src/functional/geometry/convex_hull.zig

```diff
@@ -0,0 +1,243 @@
+// Copyright 2024 TerseTS Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! This file contains the Graham algorithm to incrementally create or maintain a convex hull
+//! as well as the necessary structures, enums, and auxiliary functions. The algorithm is described
+//! in: "Mark De Berg, and Marc van Kreveld. Computational geometry: algorithms and applications.
+//! Springer Science & Business Media, 2000. https://doi.org//10.1007/978-3-540-77974-2.
+//! Convex Hulls are necessary in the implementation of the compression algorithms [1] and [2].
+//! [1] https://doi.org/10.14778/1687627.1687645 (Slide Filter).
+//! [2] https://doi.org/10.1109/TSP.2006.875394 (Optimal PLA).
+
+const std = @import("std");
+const mem = std.mem;
+const testing = std.testing;
+
+const tersets = @import("../../tersets.zig");
+const Error = tersets.Error;
+
+/// Enum for the angle's `Turn` of three consecutive points. The angle can represent a `right` or
+/// `left` turn. If there is no turn, then the points are `colinear`.
+const Turn = enum(i8) { right, left, colinear };
+
+/// Point structure to represent a point by `time` and `value`.
+const Point = struct { time: usize, value: f64 };
+
+/// Set of points used to store the upper and lower hull of the convex hull. `points` are memory
+/// slide that it's dynamically increased as `len` exceeds `max_len`. A good estimation of
+/// `max_len` can improve both execution time and memory consumption.
+pub const PointSet = struct {
```

Comment:
It is unclear why `PointSet` is implemented and used instead of `ArrayList` from Zig's standard library as it seems include the same methods that are implemented on `PointSet`. Maybe replace `PointSet` with `ArrayList`.

Reply:
Fixed. I have changed the structure to ConvexHull storing the upper and lower hulls as ArrayList(Point). I have changed all documentation accordingly.